src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -121,6 +123,9 @@ protected virtual void ReadKeyDescriptorForSigning(WsFederationConfiguration con
                 KeyInfo keyInfo = new KeyInfo();
                 keyInfo.ReadFrom(reader);
                 configuration.KeyInfos.Add(keyInfo);
+
+                var cert = new X509Certificate2(Convert.FromBase64String(keyInfo.CertificateData));
```

Comment:
We can not assume that keyInfo.CertificateData is non-null. The KeyInfo xml could be IssuerSerial etc.